src/flownet/ahm/_ahm_iteration_analytics.py

```diff
@@ -38,60 +38,57 @@ def filter_dataframe(df_in, key_filter, min_value, max_value):
     ]
 
 
-def prepare_opm_reference_data(df, str_key, n):
+def prepare_opm_reference_data(df_opm, str_key, n_real):
     """
     This function extracts data from selected columns of the Pandas dataframe
     containing data from reference simulation, rearranges it into a stacked
-    column vector preserving the original order and repeats it n times to form
-    a matrix for comparison with data from ensemble of n FlowNet simulations
+    column vector preserving the original order and repeats it n_real times to form
+    a matrix for comparison with data from ensemble of n_real FlowNet simulations
 
     Args:
-        df: is the Pandas dataframe containing data from reference simulation
+        df_opm: is the Pandas dataframe containing data from reference simulation
         str_key: is the string to select columns; column names starting with str_key
-        n: is the size of ensemble of FlowNet simulations
+        n_real: is the size of ensemble of FlowNet simulations
 
     Returns:
-        A numpy 2D array [length_data * nb_selected_columns, n] containing data
+        A numpy 2D array [length_data * nb_selected_columns, n_real] containing data
         from selected columns (i.e., quantity of interest for accuracy metric) of
-        reference simulation stacked in a column-vector and replicated into n columns
+        reference simulation stacked in a column-vector and replicated into n_real columns
     """
 
-    keys = df.keys()
-    keys = keys[df.keys().str.contains(str_key)]
-    data = np.reshape(
-        df[keys].values,
-        (df[keys].values.shape[0] * df[keys].values.shape[1], 1),
-        order="F",
-    )
-    data = np.tile(data, n)
+    keys = df_opm.keys()
+    keys = keys[df_opm.keys().str.contains(str_key)]
+    data = df_opm[keys].values
+    data = data.flatten()
+    data = np.transpose(np.tile(data, (n_real, 1)))
```

Comment:
Have not tested, but could this be simplified to e.g.
```suggestion
    data = np.transpose(np.tile(df_opm[keys].values.flatten(), (n_real, 1)))
```